server/src/main/java/org/opensearch/plugins/TracerPlugin.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugins;
+
+import org.opensearch.tracing.Telemetry;
+import org.opensearch.tracing.TracerSettings;
+
+import java.util.Map;
+import java.util.function.Supplier;
+
+/**
+ * Plugin for extending tracing related classes
+ */
+public interface TracerPlugin {
+
+    Map<String, Supplier<Telemetry>> getTelemetries(TracerSettings tracerSettings);
```

Comment:
I am not sure the `Telemetry` is even needed, it is not generic and heavily biased towards tracing,  the simpler API would be:

```
Optional<Tracer> getTracer(TracerSettings tracerSettings);
```

There could be multiple `TracerPlugin` providers, if needed, for all different kind of tracers

Reply:
I concur `Telemetry` is heavily biased towards tracing because that was our intention. We should rename it properly to clearly call out its specific to tracing (maybe TracingTelemetry or something similar). But we had added this abstraction primarily to separate Otel specific implementation being used in our tracing framework. More details [here](https://github.com/opensearch-project/OpenSearch/pull/7648#discussion_r1212888971).

The default implementation of Tracer (DefaultTrace.java) primarily keep tracks of the spans in the system using thread pool's thread context. For any Otel specific logic, it just delegates call to OtelTelemetry or OtelSpan.

Reply:
I agree with @reta and it is definitely causing some confusion.

Reply:
@suranjay please rename this plugin to `TelemetryPlugin` to reflect larger scope (this is not about just tracing anymore).

Reply:
Done